src/lib.rs

```diff
@@ -116,9 +131,25 @@ impl Preprocessor {
         GLOBALS.set(&Default::default(), || {
             let mut parsed_module = parser.parse_module()?;
 
+            let mut ident_collector = IdentCollector::new();
+            parsed_module.visit_mut_with(&mut as_folder(&mut ident_collector));
+
             let found_id = find_existing_import(&parsed_module, target_module, target_specifier);
             let had_id_already = found_id.is_some();
-            let id = found_id.unwrap_or_else(|| private_ident!(target_specifier));
+            let mut new_specifier = String::from(target_specifier);
```

Comment:
The logic for this can go inside the closure for `.unwrap_or_else` so we don’t run it unnecessarily (and we probably don’t need `private_ident!` anymore)